src/jit/emit.cpp

```diff
@@ -3048,11 +3056,12 @@ emitter::instrDesc *emitter::emitNewInstrCallDir(int        argCnt,
      */
     bool gcRefRegsInScratch = ((gcrefRegs & RBM_CALLEE_TRASH) != 0);
 
-    if  (!VarSetOps::IsEmpty(emitComp, GCvars)     ||   // any frame GCvars live
-         gcRefRegsInScratch          ||   // any register gc refs live in scratch regs
-         (byrefRegs != 0)            ||   // any register byrefs live
-         (argCnt > ID_MAX_SMALL_CNS) ||   // too many args
-         (argCnt < 0)                   ) // caller pops arguments
+    if  (!VarSetOps::IsEmpty(emitComp, GCvars)  ||                              // any frame GCvars live
+         gcRefRegsInScratch                     ||                              // any register gc refs live in scratch regs
+         (byrefRegs != 0)                       ||                              // any register byrefs live
+         (argCnt > ID_MAX_SMALL_CNS)            ||                              // too many args
+         (argCnt < 0)                                                           // caller pops arguments
+         FEATURE_UNIX_AMD64_STRUCT_PASSING_ONLY(|| (retSize1In != EA_UNKNOWN))) // There is a second ref/byref return register.
```

Comment:
I'm curious why this last condition is needed. Note that there is no check for retSize or retSizeIn in this condition. My assumption is that it is depending on the fact that a GC update will only be necessary if there are live GC refs on frame or in registers. If that is not the case for the second half of a multi-reg struct, then it would appear that there is an issue.